webapp/packages/core-utils/src/MetadataMap.test.ts

```diff
@@ -0,0 +1,237 @@
+  });
+
+  it('should return default value', () => {
+    const map = new MetadataMap<number, string>(key => `default ${key}`);
+
+    expect(JSON.stringify(map.get(1))).toStrictEqual(JSON.stringify('default 1'));
+    expect(JSON.stringify(map.get(Infinity))).toStrictEqual(JSON.stringify('default Infinity'));
+    expect(JSON.stringify(map.get(NaN))).toStrictEqual(JSON.stringify('default NaN'));
+  });
+
+  it('should return size', () => {
+    const map = new MetadataMap<number, string>();
+    map.sync([
+      [1, 'one'],
+      [2, 'two'],
+      [3, 'three'],
+    ]);
+
+    expect(map.size).toBe(3);
+  });
+
+  it('should iterate', () => {
+    const map = new MetadataMap<number, string>();
+    const data: [number, string][] = [
+      [1, 'one'],
+      [2, 'two'],
+      [3, 'three'],
+    ];
+    map.sync(data);
+
+    const items = Array.from(map);
+    expect(items).toStrictEqual([
+      [1, 'one'],
+      [2, 'two'],
+      [3, 'three'],
+    ]);
+  });
+
+  it('should iterate keys', () => {
+    const map = new MetadataMap<number, string>();
+    map.sync([
+      [1, 'one'],
+      [2, 'two'],
+      [3, 'three'],
+    ]);
+
+    const keys = Array.from(map.keys());
+    expect(keys).toStrictEqual([1, 2, 3]);
+  });
+
+  it('should iterate values', () => {
+    const map = new MetadataMap<number, string>();
+    map.sync([
+      [1, 'one'],
+      [2, 'two'],
+      [3, 'three'],
+    ]);
+
+    const values = Array.from(map.values());
+    expect(values).toStrictEqual(['one', 'two', 'three']);
+  });
+
+  it('should iterate entries', () => {
+    const map = new MetadataMap<number, string>();
+    map.sync([
+      [1, 'one'],
+      [2, 'two'],
+      [3, 'three'],
+    ]);
+
+    const entries = Array.from(map.entries());
+    expect(entries).toStrictEqual([
+      [1, 'one'],
+      [2, 'two'],
+      [3, 'three'],
+    ]);
+  });
+
+  it('should check if key exists', () => {
+    const map = new MetadataMap<number, string>();
+    map.sync([
+      [1, 'one'],
+      [2, 'two'],
+      [3, 'three'],
+    ]);
+
+    expect(map.has(1)).toBeTruthy();
+    expect(map.has(2)).toBeTruthy();
+    expect(map.has(3)).toBeTruthy();
+    expect(map.has(4)).toBeFalsy();
+  });
+
+  it('should clear', () => {
+    const map = new MetadataMap<number, string>();
+    map.sync([
+      [1, 'one'],
+      [2, 'two'],
+      [3, 'three'],
+    ]);
+
+    map.clear();
+
+    expect(map.size).toBe(0);
+  });
+
+  it('should delete', () => {
+    const map = new MetadataMap<number, string>();
+    map.sync([
+      [1, 'one'],
+      [2, 'two'],
+      [3, 'three'],
+    ]);
+
+    map.delete(2);
+
+    expect(map.has(2)).toBeFalsy();
+  });
+
+  it('should not delete not existing key', () => {
+    const map = new MetadataMap<number, string>();
+    map.sync([
+      [1, 'one'],
+      [2, 'two'],
+      [3, 'three'],
+    ]);
+
+    map.delete(4);
+
+    expect(map.size).toBe(3);
+  });
```

Comment:
can be tested as a part of "should delete"

Reply:
You also better check the snapshot of values rather than size (size also is good to check, but it's a weak check if we check only size)